src/index.ts

```diff
@@ -56,3 +71,89 @@ export function assertGoError<E extends Error>(result: GoResult<any, E>): assert
     throw new Error('Assertion failed. Expected error, but no error was thrown');
   }
 }
+
+// Adapted from:
+// https://github.com/then/is-promise
+export function isPromise(obj: any) {
+  return !!obj && (typeof obj === 'object' || typeof obj === 'function') && typeof obj.then === 'function';
+}
```

Comment:
Do we need this for anything?

Reply:
It doesn't seem to be used in Airnode or Airkeeper, remove it?

Reply:
Yes, please, do so :slightly_smiling_face: